src/databricks/labs/ucx/source_code/graph.py

```diff
@@ -589,6 +588,38 @@ def finalize(self) -> InheritedContext:
         return InheritedContext(tree, self.found)
 
 
+@dataclass
+class Lineage(abc.ABC):
+    @abc.abstractmethod
+    def as_object(self) -> dict[str, str] | list[dict[str, str]]: ...
```

Comment:
don't do union types, they are making code unmaintainable. introducing `def lineage(self) -> list[LineageAtom]:` will remove the need of `LineageList`.

Reply:
good idea, done